### DISCUSSION TITLE
How does Orient DB map between Graph and Database

### DISCUSSION THREAD

**Discussion by kaintharinder** [orientechnologies/orientdb#9773] [General]
Hi @lucafrosini ,

We are evaluating OrientDB for our application and need both Graph and Database. I have spun up an orient DB instance as graph database.

create class Page extends V
create class Instance extends V
create class Child extends E

Page ------Child (one to Many)----> Instance

I have inserted some documents ranging from 200- 250kb into both Page and Instance I see all the key-value pairs in the JSON are getting viewed as properties in a graph.

What is the mapping between graph and document here. Is it still a document in native terms and the graph viewer is showing it as properties because it is both Vertex in Graph and Document in terms of Document DB.

Next is If I traverse from page to all instances and I only need to pick suppose rid of the vertex, will it be more faster than reading the actual full document alongside the traversal. By Faster I meant will internally Orient DB first read the full record and then get me the required properties , or will it only get the rid and required properties alone from disk.

**Comment by andrii0lomakin** [comment]
Hi.
Thank you for the evaluation of OrientDB.
As for your questions:
1. `What is the mapping between graph and document here. Is it still a document in native terms and the graph viewer is showing it as properties because it is both Vertex in Graph and Document in terms of Document DB.` - it is still a document.
2. `Next is If I traverse from page to all instances and I only need to pick suppose rid of the vertex, will it be faster than reading the actual full document alongside the traversal. By Faster I meant will internally Orient DB first read the full record and then get me the required properties, or will it only get rid and required properties alone from disk.` - as for now likely, the whole document will be loaded. But it depends on the internal implementation of edge inside of a database. I suggest you to use SQL `match`  command, in such case database will internally choose the best strategy.

**Reply by kaintharinder** [reply]
Hi @laa ,

Thanks for the clarification.
I tried the match query

MATCH {class: Page, as: page, where: (name = 'Page')}.out('Child') {as: Instance}.out('Targets') {as: Segment},{as: Instance}.out('Embeds') {as: Fragment}.out('Child') {as: FragmentInstance}.out('Targets') {as: FSegment}
          RETURN page,Instance,Segment,Fragment,FragmentInstance,FSegment


currently it is returning the correct result which I need , but it is also returning the execution plan and all in the response. How can I limit it to only return the result.


"result": [
        {
            "page": "#57:1",
            "Instance": "#73:0",
            "Segment": "#89:0",
            "Fragment": "#105:0",
            "FragmentInstance": "#121:0",
            "FSegment": "#89:0"
        }
    ],
    "executionPlan": {
        "@type": "d",
        "@version": 0,

Like , I only want the values under result array.

**Comment by andrii0lomakin** [comment]
Who is '@lucafrosini' :-) ???

**Reply by kaintharinder** [reply]
I saw some articles with that name on youtube for orientdb.

**Reply by lucafrosini** [reply]
Here I am. I use OrientDB for my research project, but I'm not the right person to ask issues
